heliopy/data/mms.py

```diff
@@ -126,8 +135,15 @@ def fgm_survey(probe, starttime, endtime):
         this_relative_dir = (relative_dir /
                              str(date.year) /
                              str(date.month).zfill(2))
-        filename = 'mms{}_fgm_srvy_l2_{}{:02}{:02}_v4.18.0'.format(
-            probe, date.year, date.month, date.day)
+        
+        # Don't try to request specific versions like this - use the
+        # API to grab the most recent file.  The SDC weeps when you
+        # expect it to hold old file versions past their prime
+        datestring = '{}-{:02}-{:02}'.format(date.year,date.month,
+                                             date.day)
+        sdc_fgm_srvy = requests.get('https://lasp.colorado.edu/mms/sdc/public/files/api/v1/file_info/science?start_date='+datestring+'&end_date='+datestring+'&sc_id=mms'+probe+'&data_rate_mode=srvy&'+'instrument_id=fgm&data_level=l2')
```

Comment:
To make the code cleaner, can the main url (https://lasp.colorado.edu/mms/sdc/public/files/api/v1/file_info/science) be defined at the top of the file, and then just used as a variable here.

Reply:
Will do, I'll split into 'base url' + 'request list' for easier reworking later